code/modules/mob/living/carbon/metroid/items.dm

```diff
@@ -180,129 +180,105 @@
+	M.cores *= 3
+	qdel(src)
 
 /obj/item/slimesteroid2
 	name = "extract enhancer"
 	desc = "A potent chemical mix that will give a slime extract three uses."
 	icon = 'icons/obj/chemical.dmi'
 	icon_state = "bottle17"
 
-	/*afterattack(obj/target, mob/user , flag)
-		if(istype(target, /obj/item/slime_extract))
-			if(target.enhanced == 1)
-				to_chat(user, "<span class='warning'>This extract has already been enhanced!</span>")
-				return ..()
-			if(target.Uses == 0)
-				to_chat(user, "<span class='warning'>You can't enhance a used extract!</span>")
-				return ..()
-			to_chat(user, "You apply the enhancer. It now has triple the amount of uses.")
-			target.Uses = 3
-			target.enahnced = 1
-			qdel(src)*/
+var/list/global/golem_runes = list()
 
 /obj/effect/golemrune
-	anchored = 1
-	desc = "a strange rune used to create golems. It glows when spirits are nearby."
+	anchored = TRUE
+	desc = "A strange rune used to create golems. It glows when spirits are nearby."
 	name = "rune"
 	icon = 'icons/obj/rune.dmi'
 	icon_state = "golem"
-	unacidable = 1
+	unacidable = TRUE
 	layer = TURF_LAYER
 	var/wizardy = FALSE //if this rune can only be used by a wizard or not
 
 /obj/effect/golemrune/Initialize()
 	. = ..()
 	START_PROCESSING(SSprocessing, src)
 	announce_to_ghosts()
+	golem_runes += src
+
+/obj/effect/golemrune/Destroy()
+	. = ..()
+	golem_runes -= src
 
 /obj/effect/golemrune/process()
 	var/mob/abstract/observer/ghost
 	for(var/mob/abstract/observer/O in src.loc)
-		if(!O.client)	continue
-		if(O.mind && O.mind.current && O.mind.current.stat != DEAD)	continue
+		if(!O.client)
+			continue
+		if(O.mind && O.mind.current && O.mind.current.stat != DEAD)
+			continue
 		ghost = O
 		break
 	if(ghost && !(ghost.has_enabled_antagHUD && config.antag_hud_restricted))
 		icon_state = "golem2"
 	else
 		icon_state = "golem"
 
-/obj/effect/golemrune/attack_hand(mob/living/user as mob)
-	var/mob/abstract/observer/ghost
-	for(var/mob/abstract/observer/O in src.loc)
-		if(!O.client)	continue
-		if(O.mind && O.mind.current && O.mind.current.stat != DEAD)	continue
-		ghost = O
-		break
-
-	if(wizardy)
-		if(!user.is_wizard())
-			to_chat(user, "<span class='notice'>The rune lies silent.</span>")
-			return
-
-	if(!ghost)
-		to_chat(user, "<span class='warning'>The rune fizzles uselessly. There is no spirit nearby.</span>")
-		return
-	if(ghost.has_enabled_antagHUD && config.antag_hud_restricted)
-		to_chat(ghost, "<span class='warning'>You can not join as a golem with antagHUD on!</span>")
-		to_chat(user, "<span class='warning'>The rune fizzles uselessly. There is no spirit nearby.</span>")
-		return
-
+/obj/effect/golemrune/proc/spawn_golem(var/mob/user)
 	var/golem_type = "Adamantine Golem"
 
 	var/obj/item/stack/material/O = (locate(/obj/item/stack/material) in src.loc)
-	if(O && O.amount>=10)
+	if(O?.amount >= 10)
 		if(O.material.golem)
 			golem_type = O.material.golem
-			qdel(O)
+			O.amount -= 10
+			if(O.amount <= 0)
+				qdel(O)
```

Comment:
There's a proc for the usage stuff. `obj/item/stack/material/proc/use(var/used)`.